lib/integrations/awsoidc/eks_enroll_clusters.go

```diff
@@ -312,6 +312,11 @@ func enrollEKSCluster(ctx context.Context, log *slog.Logger, clock clockwork.Clo
 		return "", trace.BadParameter(`can't enroll EKS cluster %q - expected "ACTIVE" state, got %q.`, clusterName, eksCluster.Status)
 	}
 
+	// We can't discover private EKS clusters for cloud clients, since we know that auth server is running in our VPC.
+	if req.IsCloud && !eksCluster.ResourcesVpcConfig.EndpointPublicAccess {
+		return "", trace.AccessDenied(`can't enroll EKS cluster %q - no public access endpoint while auth is running in Teleport Cloud.`, clusterName)
```

Comment:
I would slightly change the message.
Some users won't know what `auth` means here.
```suggestion
		return "", trace.AccessDenied(`can't enroll %q because it is not accessible from Teleport Cloud, please enable endpoint public access in your EKS cluster and try again`, clusterName)
```

And possibly some documentation, but I'm not sure we can render it as a link in the UI
https://docs.aws.amazon.com/eks/latest/userguide/cluster-endpoint.html#modify-endpoint-access